internal/commands/user/list/command_test.go

```diff
@@ -0,0 +1,201 @@
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				realmClient := tc.setupClient()
+
+				cmd := &command{
+					realmClient: realmClient,
+				}
+
+				err := cmd.Handler(nil, nil)
+				assert.Equal(t, tc.expectedErr, err)
+			})
+		}
+	})
+}
+
+func TestUserListFeedback(t *testing.T) {
+	for _, tc := range []struct {
+		description    string
+		users          []realm.User
+		expectedOutput string
+	}{
+		{
+			description: "Should group the users by provider type and sort by LastAuthenticationDate",
+			users: []realm.User{
+				{
+					ID:                     "id1",
+					Identities:             []realm.UserIdentity{{ProviderType: "local-userpass"}},
+					Type:                   "type1",
+					Disabled:               false,
+					Data:                   map[string]interface{}{"email": "myEmail1"},
+					CreationDate:           1111111111,
+					LastAuthenticationDate: 1111111111,
+				},
+				{
+					ID:                     "id2",
+					Identities:             []realm.UserIdentity{{ProviderType: "local-userpass"}},
+					Type:                   "type2",
+					Disabled:               false,
+					Data:                   map[string]interface{}{"email": "myEmail2"},
+					CreationDate:           1111333333,
+					LastAuthenticationDate: 1111333333,
+				},
+				{
+					ID:                     "id3",
+					Identities:             []realm.UserIdentity{{ProviderType: "local-userpass"}},
+					Type:                   "type1",
+					Disabled:               false,
+					Data:                   map[string]interface{}{"email": "myEmail3"},
+					CreationDate:           1111222222,
+					LastAuthenticationDate: 1111222222,
+				},
+				{
+					ID:                     "id4",
+					Identities:             []realm.UserIdentity{{ProviderType: "api-key"}},
+					Type:                   "type1",
+					Disabled:               false,
+					Data:                   map[string]interface{}{"name": "myName"},
+					CreationDate:           1111111111,
+					LastAuthenticationDate: 1111111111,
+				},
+			},
+			expectedOutput: strings.Join(
+				[]string{
+					"01:23:45 UTC INFO  Provider type: local-userpass",
+					"  Email     ID   Enabled  Type   Last Authentication          ",
+					"  --------  ---  -------  -----  -----------------------------",
+					"  myEmail2  id2  true     type2  2005-03-20 15:42:13 +0000 UTC",
+					"  myEmail3  id3  true     type1  2005-03-19 08:50:22 +0000 UTC",
+					"  myEmail1  id1  true     type1  2005-03-18 01:58:31 +0000 UTC",
+					"01:23:45 UTC INFO  Provider type: api-key",
+					"  Name    ID   Enabled  Type   Last Authentication          ",
+					"  ------  ---  -------  -----  -----------------------------",
+					"  myName  id4  true     type1  2005-03-18 01:58:31 +0000 UTC",
+					"",
+				},
+				"\n",
+			),
+		},
+		{
+			description: "Should indicate no users found when none are found",
+			users:       []realm.User{},
+			expectedOutput: strings.Join(
```

Comment:
[nit] just keep this as a simple string: `"01:23:45 UTC INFO  No available users to show\n"` (and maybe move this test case to the top of the test table?)